lib/dynflow/config.rb

```diff
@@ -97,8 +97,28 @@ def queues
       5
     end
 
-    config_attr :executor, Executors::Abstract, FalseClass do |world, config|
-      Executors::Parallel.new(world, config.executor_heartbeat_interval, config.queues)
+    config_attr :executor do |world, config|
+      Executors::Parallel::Core
+    end
+
+    def validate_executor!(value)
+      accepted_executors = [Executors::Parallel::Core]
+      accepted_executors << Executors::Sidekiq::Core if defined? Executors::Sidekiq::Core
+      if value && !accepted_executors.include?(value)
+        raise ArgumentError, "Executor #{value} is expected to be one of #{accepted_executors.inspect}"
+      end
+    end
+
+    # does the work represent some process-wide role (important for sidekiq-based deployments)
+    config_attr :process_role do |world, config|
```

Comment:
Why is this introduced? Why do we need to separate the worlds? I do not see any particular reason sidekiq world can't be shared if required. This could lead to completly different flows for sidekiq and without, if we start branch the code depending on this (and if we won't why we need it).
If we are at it, maybe we could do it for all worlds, not just for sidekiq?